load_text only creates populations if none given

hyanwong opened the pull request:
Fixes #1909

# PR Checklist:

- [x] Tests that fully cover new/changed functionality.
- [x] Documentation including tutorial content if appropriate.
- [x] Changelogs, if there are API changes.

jeromekelleher commented:
Looks good. It would be nice to add a couple of minimal tests like (paraphrased)
```python
def test_parse_text_no_populations(self):
    nodes = """\
       is_sample  time  population
        1                0      2
     """
     tables = parse_text(nodes)
     assert len(tables.nodes) == 1
     assert len(tables.populations) == 3

def test_parse_text_populations(self):
       populations = """\
       metadata
       metadata1
       metadata2
       """
       tables = parse_text(nodes)     
       assert len(tables.populations) == 2
       assert tables.popualtions[0].metadata == "metadata1"
       assert tables.popualtions[0].metadata == "metadata2"
```

This will exercise both paths. I'm becoming a big fan of small obvious tests, they're so much more effective than the big elaborate code-based approaches that I used to favour.

hyanwong commented:
> Looks good. It would be nice to add a couple of minimal tests...

Good call. This exposed another bug: we were forgetting to strip the "\n" from each line when doing strict text reading. I've added this in, but apart from the test you suggested above, which only applied to the population table, this change doesn't have an associated unit test.

(I've also changed the `strip("\n")` to `rstrip("\n")` which is a bit more specific.

jeromekelleher reviewed (approved):
LGTM
